src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java

```diff
@@ -192,7 +230,7 @@ private void replaceIndex(final ActionRequest request, final String oldIndexName
 
         // CreateIndexRequest
         if (request instanceof CreateIndexRequest) {
-            ((CreateIndexRequest) request).index(newIndexName);
+            ((CreateIndexRequest) request).index(newIndexName + "_0").alias(new Alias(newIndexName));
```

Comment:
Just out of curiosity, does CreateIndexReqeust come to here at all? I tried doing same thing, but it didn't work.

Reply:
`CreateIndexRequest ` may be submitted using REST API, it won't be submitted as part of regular Kibana workflow. The change is required for the consistent behavior in index re-write.

Reply:
That makes sense. Thanks for explanatino!